Update .gitignore.tmpl

NebiSarigul opened the pull request:
## Description

When building with Android Studio, .apk, .aab and .json files are added to this directory, which causes it to be uploaded to git. I created a new project with "flutter create". Then I opened the project's android directory with Android Studio. I created the app bundle with the Genarate signed bundle option from the Build menu. The app-release.aab file occurred in the /android/app/release/ directory.

NebiSarigul committed:
Update .gitignore.tmpl

When building with Android Studio, .apk, .aab and .json files are added to this directory, which causes it to be uploaded to git. I created a new project with "flutter create". Then I opened the project's android directory with Android Studio. I created the app bundle with the Genarate signed bundle option from the Build menu. The app-release.aab file occurred in the /android/app/release/ directory.

jonahwilliams commented on packages/flutter_tools/templates/app/.gitignore.tmpl:
```diff
@@ -39,3 +39,7 @@ app.*.symbols
 # Obfuscation related
 app.*.map.json
+
+**/android/app/debug
```
This will ignore every single folder that matches this pattern. Instead, you should only ignore the exact folders, like `/android/app/debug`. 

Also please leave a comment describing what this is for

NebiSarigul committed:
Update .gitignore.tmpl

When building with Android Studio, .apk, .aab and .json files are added to this directory, which causes it to be uploaded to git. I created a new project with "flutter create". Then I opened the project's android directory with Android Studio. I created the app bundle with the Genarate signed bundle option from the Build menu. The app-release.aab file occurred in the /android/app/release/ directory.

jonahwilliams commented on packages/flutter_tools/templates/app/.gitignore.tmpl:
```diff
@@ -39,3 +39,8 @@ app.*.symbols
 # Obfuscation related
 app.*.map.json
+
+# Android related
```
How about "Android Studio will place build artifacts here" or something similar?

NebiSarigul replied:
Good idea. i changed.

NebiSarigul committed:
Update .gitignore.tmpl

Comment changed.

jonahwilliams reviewed (approved):
LGTM
